Sources/LocalReceiptParsing/Builders/ASN1ContainerBuilder.swift

```diff
@@ -84,7 +83,8 @@ private extension ASN1ContainerBuilder {
         return asn1Identifier
     }
 
-    func extractLength(data: ArraySlice<UInt8>) throws -> ASN1Length {
+    func extractLengthAndInternalContainers(data: ArraySlice<UInt8>,
+                                            isConstructed: Bool) throws -> (ASN1Length, [ASN1Container]) {
```

Comment:
this is awkward and I don't love it, but I couldn't separate the two steps, because: 
- for definite-length receipts, internal containers are determined by the receipt length
- for indefinite-length receipts, the actual length is determined by the internal containers' length
- I didn't want to generate internal containers twice

Alternatively I could have had the same old logic, but with a special case for indefinite receipts, where we initially set the length to zero, and then after building internal containers, create a new instance of ASN1Length with the correct value. 

That felt confusing and odd so I figured it's somewhat better to just determine both in a single step